src/domain/flow-system/services/src/flow/flow_service_impl.rs

```diff
@@ -271,27 +273,31 @@ impl FlowServiceImpl {
                 trigger_time: start_time,
             }),
             FlowTriggerContext::Unconditional,
+            ConfigSnapshot::default(),
         )
         .await
     }
 
-    #[tracing::instrument(level = "trace", skip_all, fields(?flow_key, %flow_id))]
+    #[tracing::instrument(level = "trace", skip_all, fields(?flow))]
     async fn enqueue_dependent_flows(
         &self,
         input_success_time: DateTime<Utc>,
-        flow_key: &FlowKey,
-        flow_id: FlowID,
+        flow: &Flow,
         flow_result: &FlowResult,
     ) -> Result<(), InternalError> {
-        if let FlowKey::Dataset(fk_dataset) = flow_key {
+        if let FlowKey::Dataset(fk_dataset) = &flow.flow_key {
             // Extract a list of downstream 1 level datasets
+            let flow_manual_trigger_initiator = flow
+                .try_get_manual_trigger()
+                .map(|trigger| trigger.initiator_account_id);
+
             let dependent_dataset_ids: Vec<_> = self
-                .dependency_graph_service
-                .get_downstream_dependencies(&fk_dataset.dataset_id)
-                .await
-                .int_err()?
-                .collect()
-                .await;
+                .permissions_plugin
+                .get_account_downstream_dependencies(
+                    &fk_dataset.dataset_id,
+                    flow_manual_trigger_initiator,
+                )
```

Comment:
Note that this requirement to trigger only own accounts is specific for compacting, but is not the same for update flows - an update of root should trigger all derived, regardless of who owns them